cmd/thanos/compact.go

```diff
@@ -495,6 +512,9 @@ func (cc *compactConfig) registerFlag(cmd *kingpin.CmdClause) {
 	cmd.Flag("retention.resolution-1h", "How long to retain samples of resolution 2 (1 hour) in bucket. Setting this to 0d will retain samples of this resolution forever").
 		Default("0d").SetValue(&cc.retentionOneHr)
 
+	cmd.Flag("add-hashes", "Add hashes to meta file after producing new blocks which permits avoiding downloading some files twice albeit at some performance cost.").
```

Comment:
`downloading some file twice`  is vague - does it mean Thanos downloads files couple of times without checking? Let's maybe not specify what's the purpose - there are many reasons. Mainly to ensure data integrity.

Can we at some point share what is that performance cost and during what moment? ( :